src/main/java/com/example/onlinebookshop/mapper/ShoppingCartMapper.java

```diff
@@ -0,0 +1,36 @@
+package com.example.onlinebookshop.mapper;
+
+import com.example.onlinebookshop.config.MapperConfig;
+import com.example.onlinebookshop.dto.cartitem.response.CartItemDto;
+import com.example.onlinebookshop.dto.shoppingcart.response.ShoppingCartDto;
+import com.example.onlinebookshop.entities.ShoppingCart;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+
+@Mapper(config = MapperConfig.class)
+public interface ShoppingCartMapper {
+    @Mapping(target = "userId", ignore = true)
```

Comment:
It could be simpler without setUserId
```suggestion
    @Mapping(source = "user.id", target = "userId")
```